packages/react-native-web/src/exports/Modal/index.js

```diff
@@ -1,2 +1,251 @@
-export default UnimplementedView;
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import React, { forwardRef, useRef, useCallback, useMemo, useEffect } from 'react';
+
+import { canUseDOM } from 'fbjs/lib/ExecutionEnvironment';
+
+import View from '../View';
+import StyleSheet from '../StyleSheet';
+
+import type { ModalProps } from './types';
+
+import ModalPortal from './ModalPortal';
+import ModalAnimation from './ModalAnimation';
+import FocusBracket from './FocusBracket';
+
+function attemptFocus(element: any) {
+  try {
+    element.focus();
+  } catch (e) {
+    // Do nothing
+  }
+
+  return document.activeElement === element;
+}
+
+function focusFirstDescendant(element: any) {
+  for (let i = 0; i < element.childNodes.length; i++) {
+    const child = element.childNodes[i];
+    if (attemptFocus(child) || focusFirstDescendant(child)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function focusLastDescendant(element: any) {
+  for (let i = element.childNodes.length - 1; i >= 0; i--) {
+    const child = element.childNodes[i];
+    if (attemptFocus(child) || focusLastDescendant(child)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+let uniqueModalIdentifier = 0;
+
+const visibleModalStack = [];
+
+function isTopModal(modalId) {
+  if (visibleModalStack.length === 0) {
+    return false;
+  }
+
+  return visibleModalStack[visibleModalStack.length - 1] === modalId;
+}
+
+const Modal = forwardRef<ModalProps, *>((props, forwardedRef) => {
+  const {
+    visible,
+    animated,
+    animationType,
+    transparent,
+    children,
+    onShow: onShowProp,
+    onDismiss: onDismissProp,
+    onRequestClose: onRequestCloseProp
+  } = props;
+
+  const modalElementRef = useRef();
+
+  // Sync the internal ref we track into the forwarded ref
+  useEffect(() => {
+    if (!forwardedRef) {
+      return;
+    }
+
+    if (typeof forwardedRef === 'function') {
+      forwardedRef(modalElementRef.current);
+    } else {
+      forwardedRef.current = modalElementRef.current;
+    }
+  }, [forwardedRef])
+
+
+  // Set a unique model identifier so we can correctly route
+  // dismissals and check the layering of modals.
+  const modalId = useMemo(() => uniqueModalIdentifier++, []);
+
+  // Ref used to track trapping of focus and to prevent focus from leaving a modal
+  // for accessibility reasons per W3CAG.
+  const focusRef = useRef<{ trapFocusInProgress: boolean, lastFocusedElement: ?HTMLElement }>({
+    trapFocusInProgress: false,
+    lastFocusedElement: null
+  });
+
+  const onDismiss = useCallback(() => {
+    // When we dismiss we can't assume that we're dismissing the
+    // top element in the stack - so search the stack and remove
+    // ourselves from it if need be.
+    if (visibleModalStack.includes(modalId)) {
```

Comment:
IE11 doesn't support `.includes` so let's use an index check

Reply:
Ah! Definitely, will do. Was there a check I should have run to detect this statically?